src/tile.cpp

```diff
@@ -825,36 +828,36 @@ Tile* Tile::queryDestination(int32_t&, const Thing&, Item** destItem, uint32_t&
 	}
 
 	if (!destTile) {
-		destTile = this;
+		destTile = getTile();
 	} else {
 		flags |= FLAG_NOLIMIT; // Will ignore that there is blocking items/creatures
 	}
 
 	if (destTile) {
-		Thing* destThing = destTile->getTopDownItem();
+		auto destThing = destTile->getTopDownItem();
```

Comment:
Could this be `const auto`
```suggestion
		const auto destThing = destTile->getTopDownItem();
```